core/src/main/java/nl/nn/adapterframework/util/FileUtils.java

```diff
@@ -262,20 +262,25 @@ public static File createTempFile() throws IOException {
 		return createTempFile(null);
 	}
 	public static File createTempFile(String suffix) throws IOException {
-		return createTempFile(null,null);
+		return createTempFile(null,suffix);
 	}
 	public static File createTempFile(String prefix, String suffix) throws IOException {
-		String directory=AppConstants.getInstance().getResolvedProperty("upload.dir");
+		String directoryName=AppConstants.getInstance().getResolvedProperty("upload.dir");
+		if (directoryName == null) {
+			throw new IOException("Property [upload.dir] is not specified");
+		}
+		File directory = new File(directoryName);
+		if (!directory.exists() && (!directory.mkdirs())) {
+			throw new IOException("Unable to create [upload.dir] at location [" + directoryName + "]");
+		}
 		if (StringUtils.isEmpty(prefix)) {
 			prefix="ibis";
 		}
 		if (StringUtils.isEmpty(suffix)) {
 			suffix=".tmp";
 		}
 		if (log.isDebugEnabled()) log.debug("creating tempfile prefix ["+prefix+"] suffix ["+suffix+"] directory ["+directory+"]");
-		File tmpFile = File.createTempFile(prefix, suffix, new File(directory));
-		tmpFile.deleteOnExit();
```

Comment:
Is er eigenlijk een codestyle-rule te schrijven ergens die build laat falen op gebruik van `File.deleteOnExit()` anders dan in test-code?

Reply:
Ja zal ik doen.